src/com/xilinx/rapidwright/design/tools/RelocationTools.java

```diff
@@ -0,0 +1,293 @@
+import com.xilinx.rapidwright.edif.EDIFNetlist;
+import com.xilinx.rapidwright.edif.EDIFTools;
+import com.xilinx.rapidwright.util.Pair;
+
+/**
+ * A collection of tools to help relocate designs.
+ *
+ * @author eddieh
+ *
+ */
+public class RelocationTools {
+
+    /**
+     * Relocate all SiteInsts (and all associated PIPs) belonging to the logical Cell at
+     * instanceName in-place by tileColOffset/tileRowOffset tiles.
+     *
+     * Should a SiteInst contain a mix of physical Cells belonging to and outside of
+     * instanceName then this function will fail.
+     *
+     * @param design Parent design
+     * @param instanceName Full hierarchical instance name to logical cell
+     *                     (empty for top cell)
+     * @param tileColOffset Relocate this number of tile columns (X axis)
+     * @param tileRowOffset Relocate this number of tile rows (Y axis)
+     * @return True if successful, false otherwise.
+     */
+    public static boolean relocate(Design design,
+                                   String instanceName,
+                                   int tileColOffset,
+                                   int tileRowOffset) {
+        EDIFNetlist netlist = design.getNetlist();
+        EDIFHierCellInst instanceCell = netlist.getHierCellInstFromName(instanceName);
+        if (instanceCell == null) {
+            System.out.println("ERROR: Logical cell with instance name '" + instanceName + "' not found");
+            return false;
+        }
+
+        Set<Cell> cells = new HashSet<>();
+        Set<SiteInst> siteInsts = new HashSet<>();
+        for (EDIFHierCellInst leaf : netlist.getAllLeafDescendants(instanceCell)) {
+            String leafName = leaf.getCellName();
+            if (leafName.equals("GND") || leafName.equals("VCC")) {
+                continue;
+            }
+
+            Cell c;
+            // FIXME: Workaround the CFGLUT5 macro not being expanded correctly
+            if (leafName.equals("CFGLUT5")) {
+                Cell c1 = design.getCell(leaf.getFullHierarchicalInstName() + EDIFTools.EDIF_HIER_SEP + "S1");
+                Cell c2 = design.getCell(leaf.getFullHierarchicalInstName() + EDIFTools.EDIF_HIER_SEP + "S2");
+                assert(c1 != null && c2 != null);
+                assert(c1.getSiteInst() == c2.getSiteInst());
+                cells.add(c1);
+                c = c2;
+            }
+            else {
+                c = design.getCell(leaf.getFullHierarchicalInstName());
+                if (c == null) {
+                    System.out.println("WARNING: Could not find physical cell corresponding to logical cell '" +
+                            leaf.getFullHierarchicalInstName() + "'; ignoring");
+                    continue;
+                }
+            }
+
+            cells.add(c);
+
+            SiteInst si = c.getSiteInst();
+            if (si == null) {
+                continue;
+            }
+
+            // TODO: Is this whitelist sufficient?
```

Comment:
@clavin-xlnx

Reply:
Take a look at `Utils.java` (`util` package) and `EnumSet`